src/main/scala/nl/biopet/tools/multigenicsearch/MultigenicSearch.scala

```diff
@@ -0,0 +1,287 @@
+    * @return Output dataset
+    */
+  def initCombinations(variants: Dataset[SingleSamples])(
+      implicit spark: SparkSession): Dataset[Combination] = {
+    import spark.implicits._
+
+    val singleSamples1 = variants
+      .withColumnRenamed("index", "index1")
+      .withColumnRenamed("variantSamples", "samples1")
+      .as("singleSamples1")
+    val singleSamples2 = variants
+      .withColumnRenamed("index", "index2")
+      .withColumnRenamed("variantSamples", "samples2")
+      .as("singleSamples2")
+
+    singleSamples1
+      .join(singleSamples2, $"index1" < $"index2")
+      .withColumn("indexes", array($"index1", $"index2"))
+      .withColumn("samples", array($"samples1", $"samples2"))
+      .select("indexes", "samples")
+      .as[Combination]
+  }
+
+  /**
+    *
+    * @param variants Variants to add
+    * @param current Current combinations
+    * @param spark Implicit spark session
+    * @return Output dataset
+    */
+  def addCombination(variants: Dataset[SingleSamples],
+                     current: Dataset[Combination])(
+      implicit spark: SparkSession): Dataset[Combination] = {
+    import spark.implicits._
+
+    val addIndex: UserDefinedFunction = udf(
+      (array: Seq[Long], value: Long) => array :+ value)
+    val addSamples: UserDefinedFunction = udf(
+      (array: Seq[Seq[Boolean]], value: Seq[Boolean]) => array :+ value)
+
+    current
+      .join(variants, $"indexes" (size($"indexes") - 1) < $"index")
+      .withColumnRenamed("indexes", "indexes_old")
+      .withColumnRenamed("samples", "samples_old")
+      .withColumn("indexes", addIndex($"indexes_old", $"index"))
+      .withColumn("samples", addSamples($"samples_old", $"variantSamples"))
+      .select("indexes", "samples")
+      .as[Combination]
+  }
+
+  /**
+    * Writing result as partitioned json file
+    * @param variants All variants
+    * @param combinations Combinations to write
+    * @param maxNonVariants Max allowed non-variant calls per sample
+    * @param sampleNumber Tot number of samples
+    * @param sampleFraction Fraction of samples that should pass
+    * @param outputDir Output dir to write files, should not yet exist
+    * @return
+    */
+  def writeResult(variants: Dataset[SingleVariantWithIndex],
+                  combinations: Dataset[Combination],
+                  maxNonVariants: Int,
+                  sampleNumber: Int,
+                  sampleFraction: Double,
+                  outputDir: File): Future[Unit] = {
+    val filterCombinations = filterMaxNonVariants(combinations,
+                                                  maxNonVariants,
+                                                  sampleNumber,
+                                                  sampleFraction)
+
+    Future {
+      filterCombinations.write.json(outputDir.getAbsolutePath)
+    }
+  }
+
+  def descriptionText: String =
+    """
+      |This tool will try to find multigenic events with different sizes.
```

Comment:
An extra line about this tool's definition of 'multigenic' would be nice to have.